src/app/state/accountHistorySlice.ts

```diff
@@ -170,15 +211,30 @@ export const accountHistorySlice = createSlice({
     resetSelectedOrdersToCancel: (state) => {
       state.selectedOrdersToCancel = {};
     },
+    setHideOtherPairs: (state, action: PayloadAction<boolean>) => {
+      state.hideOtherPairs = action.payload;
+    },
   },
 
   extraReducers: (builder) => {
-    builder.addCase(fetchAccountHistory.fulfilled, (state, action) => {
-      state.orderHistory = action.payload.data.orders;
-    });
-    builder.addCase(batchCancel.fulfilled, (state) => {
-      state.selectedOrdersToCancel = {};
-    });
+    builder
+      .addCase(fetchAccountHistory.fulfilled, (state, action) => {
+        state.orderHistory = action.payload.data.orders;
+      })
+      .addCase(fetchAccountHistory.pending, (state) => {
+        state.loading = true;
+        state.error = null;
+      })
+      .addCase(fetchAccountHistory.rejected, (state, action) => {
+        state.loading = false;
+        state.error = action.error.message || "Failed to fetch account history";
+      })
```

Comment:
Did you intend to add those cases for `fetchAccountHistoryAllPairs`?

Reply:
<img width="713" alt="Screenshot 2024-07-25 at 21 20 58" src="https://github.com/user-attachments/assets/56b287db-4024-4cf9-883f-8a6ac77b09fd">

I added those cases in response to your comment above, I thought they are for `fetchAccountHistory` 😅 
I think I get it now, so you just sent all 3 cases as a reference, and in our case we only need to catch the error inside the extraReducers - in the `fetchAccountHistoryAllPairs.rejected` (not `fetchAccountHistory.rejected` as I did)

Reply:
Yes exactly! Sorry for the confusion!